Essentials/src/main/java/com/earth2me/essentials/EssentialsUpdateChecker.java

```diff
@@ -0,0 +1,258 @@
+package com.earth2me.essentials;
+
+import com.google.common.base.Charsets;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import static com.earth2me.essentials.I18n.tl;
+
+public final class EssentialsUpdateChecker {
+    private static final String REPO = "EssentialsX/Essentials";
+    private static final String BRANCH = "2.x";
+
+    private static final String versionIdentifier;
+    private static final String versionBranch;
+    private static final boolean devBuild;
+    private static long lastFetchTime = 0;
+    private static CompletableFuture<UpdateToken> pendingDevFuture;
+    private static CompletableFuture<UpdateToken> pendingReleaseFuture;
+    private static String latestRelease = null;
+    private static UpdateToken cachedDev = null;
+    private static UpdateToken cachedRelease = null;
+
+    static {
+        String identifier = "INVALID";
+        String branch = "INVALID";
+        boolean dev = false;
+        final List<String> versionStr = new BufferedReader(new InputStreamReader(Objects.requireNonNull(EssentialsUpdateChecker.class.getClassLoader().getResourceAsStream("release")), StandardCharsets.UTF_8)).lines().collect(Collectors.toList());
+        if (versionStr.size() == 2) {
+            if (versionStr.get(0).matches("\\d+\\.\\d+\\.\\d+-dev\\+\\d\\d-[0-9a-f]{7,40}")) {
+                identifier = versionStr.get(0).split("-")[2];
+                dev = true;
+            } else {
+                identifier = versionStr.get(0);
+            }
+            branch = versionStr.get(1);
+        }
```

Comment:
```suggestion
        final InputStream inputStream = EssentialsUpdateChecker.class.getClassLoader().getResourceAsStream("release");
        if (inputStream != null) {
            final List<String> versionStr = new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8)).lines().collect(Collectors.toList());
            if (versionStr.size() == 2) {
                if (versionStr.get(0).matches("\\d+\\.\\d+\\.\\d+-dev\\+\\d\\d-[0-9a-f]{7,40}")) {
                    identifier = versionStr.get(0).split("-")[2];
                    dev = true;
                } else {
                    identifier = versionStr.get(0);
                }
                branch = versionStr.get(1);
            }
        }
```

Afaik `Objects.requireNonNull` in the static block here will cause the class to not initialize, so EssentialsX won't load if the release file is missing.